From e4841b9dd82045e99f043e88edb5ecef9bc5954d Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Tue, 10 Mar 2015 15:26:46 +0800
Subject: [PATCH 0103/5242] MLK-10394 usb: chipidea: otg_fsm: turn off vbus
 when remove driver

commit  5e60d35c782c01ff17daaa4fbadebe38fb7a6e21 from
https://source.codeaurora.org/external/imx/linux-imx.git

Currently if remove controller driver while vbus is on, vbus turn off
is missing, thus next time loading controller driver will turn on vbus
again but will not generate AVV irq, finally otg fsm state is mess and
cannot work as host correctly.

Signed-off-by: Li Jun <jun.li@freescale.com>
(cherry picked from commit c810ab9697dd912ba2a928659879ace188d31c10)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/otg_fsm.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/chipidea/otg_fsm.c b/drivers/usb/chipidea/otg_fsm.c
index ba79f43..59ce2d4 100644
--- a/drivers/usb/chipidea/otg_fsm.c
+++ b/drivers/usb/chipidea/otg_fsm.c
@@ -860,6 +860,10 @@ int ci_hdrc_otg_fsm_init(struct ci_hdrc *ci)
 
 void ci_hdrc_otg_fsm_remove(struct ci_hdrc *ci)
 {
+	/* Turn off vbus if vbus is on */
+	if (ci->fsm.a_vbus_vld)
+		otg_drv_vbus(&ci->fsm, 0);
+
 	sysfs_remove_group(&ci->dev->kobj, &inputs_attr_group);
 }
 
-- 
1.7.9.5

